core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -119,6 +120,11 @@ enum Route {
         LoadTableResponse.class),
     LOAD_TABLE(
         HTTPMethod.GET, "v1/namespaces/{namespace}/tables/{table}", null, LoadTableResponse.class),
+    REGISTER_TABLE(
+        HTTPMethod.POST,
+        "v1/namespaces/{namespace}/tables/register",
+        RegisterTableRequest.class,
+        LoadTableResponse.class),
```

Comment:
do we need a new response type? Also, this is not inline with the open-api spec response type.